### PR TITLE
Use the RoCM/HIP device to accelerate certain DPLASMA kernels

### PR DESCRIPTION
**Pull request by abouteiller**
This PR adds RoCM enabled kernels to the GEMM, PORTF and memory-aware GEMM operations.

### CODE REVIEW

**Comment by bosilca** [comment]
please squash to fewer commits.

**Comment by abouteiller** [comment]
This is in ready to merge state beside the 'squash to less commits'.

**Comment by abouteiller** [comment]
I decided to go for a squash merge, this is ready for final review @bosilca @devreal 

Error in ctest is #115, preexisting and unrelated to hip.

File: src/dplasmaaux.h [hunk 1]
```diff
@@ -109,8 +109,7 @@ extern void *dplasma_pcomm;
 #define dplasma_error(__func, __msg) do { fprintf(stderr, "%s: %s\n", (__func), (__msg)); } while(0)
 #endif /* defined(DPLASMA_DEBUG) */
 
-#if defined(DPLASMA_HAVE_CUDA)
 #include "dplasmaaux_cuda.h"
```
Comment (bosilca) [inline src/dplasmaaux.h:112]: Why are these headers not protected ?

File: src/zgemm_TN.jdf [hunk 2]
```diff
@@ -169,8 +169,8 @@ BODY [type=CUDA
       B.size=%{ return descB->mb*descB->nb*parsec_datadist_getsizeoftype(descB->mtype);%}
       C.size=%{ return descC->mb*descC->nb*parsec_datadist_getsizeoftype(descC->mtype);%}
       A.dc=ddescA B.dc=ddescB C.dc=ddescC
-      stage_in=stage_in_lapack
-      stage_out=stage_out_lapack]
+      stage_in=dplasma_cuda_lapack_stage_in
+      stage_out=dplasma_cuda_lapack_stage_out]
```
Comment (bosilca) [inline src/zgemm_TN.jdf:173]: Why there is no HIP chore for the normal GEMM (aka. not the summa version) ?

**Review by bosilca — COMMENTED** [review]
Why only certain kernels are HIP-enabled instead of having all CUDA-enabled kernel also be HIP-enabled ?

File: src/dplasmaaux.h (hunk 1 above)
Comment (abouteiller) [reply src/dplasmaaux.h:112]: protection is self contained in the header itself

File: src/zgemm_TN.jdf (hunk 2 above)
Comment (abouteiller) [reply src/zgemm_TN.jdf:173]: Not all kernels are implemented https://github.com/ICLDisco/dplasma/issues/98

File: src/dplasmaaux.h (hunk 1 above)
Comment (bosilca) [reply src/dplasmaaux.h:112]: We had proper protection in place avoiding to include unnecessary files. What justify the need to made this change ?

File: share/help-dplasma.txt [hunk 3]
```diff
@@ -1,8 +1,9 @@
...
+There was not enough memory available on a GPU device
 while trying to allocate a %s handle to manage tasks on
-this device, or another CUDA device on the node. The
+this device, or another GPU device on the node. The
 PaRSEC runtime system may be configured to reserve too
-much memory on CUDA devices. Try reducing the amount of
+much memory on GPU devices. Try reducing the amount of
 reserved memory by setting the PaRSEC MCA parameter
-'device_cuda_memory_use' to a lower value.
+'device_cuda_memory_use' (or similar for the type of
```
Comment (devreal) [inline share/help-dplasma.txt:8]: Should we mention `device_hip_memory_use` here? (if that's the right name?)

File: src/dplasmaaux_hip.h [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    assert( (diag == dplasmaNonUnit) || (diag == dplasmaUnit) );               \
+    diag = (diag == dplasmaNonUnit) ? HIPBLAS_DIAG_NON_UNIT : HIPBLAS_DIAG_UNIT;
+
+#define dplasma_hipblas_fill(fill)                                                   \
+    assert( (fill == dplasmaLower) || (fill == dplasmaUpper) );                      \
+    fill = (fill == dplasmaLower) ? HIPBLAS_FILL_MODE_LOWER : HIPBLAS_FILL_MODE_UPPER;
+
+#if defined(PRECISION_z) || defined(PRECISION_c)
+#define dplasma_hipblas_op(trans)                 \
+    assert( (trans == dplasmaNoTrans) || (trans == dplasmaTrans) || (trans == dplasmaConjTrans) ); \
```
Comment (devreal) [inline src/dplasmaaux_hip.h:32]: I really dislike these being macros. Inline functions would be more appropriate and less error prone but if the CUDA variants do the same then so be it.

**Pull request merged**